src/ipx.ts

```diff
@@ -191,32 +189,57 @@ class IPX {
     }
 
     // Read buffer from input
-    let data = await this.get(info.src, info.adapter)
-    let sharp = Sharp(data)
-
-    if (!this.skipOperations(info)) {
-      sharp = this.applyOperations(sharp, info)
-      data = await sharp.toBuffer()
-    }
-
-    if (info.format.match(/\.json$/)) {
-      const metadata = await sharp.metadata()
-      data = Buffer.from(JSON.stringify({
-        data: `data:image/${metadata.format};base64,${data.toString('base64')}`,
-        width: metadata.width,
-        height: metadata.height,
-        size: metadata.size
-      }))
+    const originalBuffer = await this.get(info.src, info.adapter)
+    let sharp = Sharp(originalBuffer)
+    let result = null
+    const operationsContext: any = {}
+
+    switch (info.format) {
+      case 'sqip':
+        result = await this.generateSQIP(originalBuffer)
+        break
+      case 'svg':
+        result = originalBuffer
+        break
+      default:
+        sharp = this.applyOperations(sharp, info, operationsContext)
+        result = await sharp.toBuffer()
+        if (info.format === 'meta') {
+          result = await this.getMetadata(sharp, info, operationsContext)
+        }
     }
 
     // Put data into cache
     try {
-      await this.cache!.set(info.cacheKey, data)
+      await this.cache!.set(info.cacheKey, result)
     } catch (e) {
       consola.error(e)
     }
 
-    return data
+    return result
+  }
+
+  async getMetadata (sharp: Sharp.Sharp, info: IPXImageInfo, operationsContext: any) {
+    let result = await sharp.toBuffer()
+    const metadata = await sharp.metadata()
+    let metadataFormat = metadata.format
+    const meta = {
+      data: undefined,
+      width: metadata.width,
+      height: metadata.height,
+      bytes: metadata.size
+    }
+    if (operationsContext.metaDataType === 'sqip') {
+      result = await this.generateSQIP(result)
+      metadataFormat = 'sqip'
+    }
+    if (operationsContext.metaDataEncode === 'url') {
+      meta.data = `/${info.adapter}/${metadataFormat}/w_30/${info.src}`
```

Comment:
I think we may simply construct full-url to data in client-side. This will introduce duplicate logic in two places